src/services/EdFi.DataManagementService.Api.Tests.E2E/Features/SchoolYearValidation.feature

```diff
@@ -0,0 +1,203 @@
+@SchoolYear
+Feature: School Year Reference Validation
+  Reference validation on School Years
+
+  """
+  @StudentEducationOrganizationAssociation Hook that will create the following data for the required tests
+  createStudent(studentUniqueId)
+  createSchool(schoolId)
+  createCourse(courseCode, schoolId)
+  createSession(schoolId, schoolYear, sessionName)
+  createCourseOffering(localCourseCode, courseCode, schoolId, schoolYear, sessionName)
+  createSection(localCourseCode, schoolId, schoolYear, sessionName, sectionIdentifier)
+  """
+
+  @Ignore @StudentEducationOrganizationAssociation
+  Scenario: Try creating a resource using a valid school year
+    When sending a POST request to "/ed-fi/studentSectionAssociation" with body
+    """
+        {
+        "beginDate": "2024-04-25",
+        "sectionReference": {
+            "localCourseCode": "string",
+            "schoolId": 0,
+            "schoolYear": 0,
+            "sectionIdentifier": "string",
+            "sessionName": "string",
+        },
+        "studentReference": {
+            "studentUniqueId": "string",
+        }
+        }
+    """
+    Then the response code is 201
+    And the response body is
+    """
+    """
+
+  @Ignore @StudentEducationOrganizationAssociation
+  Scenario: Try creating a resource using an invalid school year
+    When sending a POST request to "/ed-fi/studentSectionAssociation" with body
```

Comment:
`Section` does not depend directly on `SchoolYear`: it is related to `CourseOffering`. This test is actually validating what happens when a `CourseOffering` does not exist.  

The [Handbook for Data Standard 5.0](https://schema.ed-fi.org/datahandbook-v500/index.html#/) is a good place to explore these relationships. Still, it is a little difficult to tell how `SchoolYear` ended up in `Section`. I have been working on this long enough that I know. You would need to look at the Handbook and search "Section". Then you would see that `SchoolYear` is not there. You would have to start guessing where it is. So maybe that is not the best resource after all.

The [main Data Standard 5.0 documentation](https://edfi.atlassian.net/wiki/spaces/EFDS5/pages/26707002/Unifying+Data+Model+-+v5+Model+Reference) has model diagrams, though you need to know where to look for them. I found `Section` under [Teaching and Learning Domain](https://edfi.atlassian.net/wiki/spaces/EFDS5/pages/26707383/Teaching+and+Learning+Domain+-+Model+Diagrams). If you follow the lines carefully then you can trace that `Section` "has associated" `CourseOffering`, which "has associated" `Session`, and `Session` has a property called `SchoolYear`.

Conclusion: to test proper recognition of school years, you'll need to POST a `Session`.

```json
{
    "schoolReference": {
      "schoolId": 255901001
    },
    "schoolYearTypeReference": {
      "schoolYear": 2022
    },
    "sessionName": "2021-2022 Fall Semester",
    "beginDate": "2021-08-23",
    "endDate": "2021-12-17",
    "termDescriptor": "uri://ed-fi.org/TermDescriptor#Fall Semester",
    "totalInstructionalDays": 81,
 }
```

Here's a sample error message when the `schoolYear` does not exist:

```json
{
  "detail": "The referenced 'SchoolYearType' resource does not exist.",
  "type": "urn:ed-fi:api:conflict:invalid-reference",
  "title": "Resource Not Unique Conflict due to invalid-reference",
  "status": 409,
  "correlationId": "5c0bd5a8-44da-4559-8702-1e2ef83c4f23"
}
```

Reply:
@msilesgap please see the comment above.